lib/Twig/Environment.php

```diff
@@ -573,7 +573,13 @@ public function compile(Twig_Node $node)
     public function compileSource($source, $name = null)
     {
         try {
-            return $this->compile($this->parse($this->tokenize($source, $name)));
+            $compiled = $this->compile($this->parse($this->tokenize($source, $name)), $source);
+
+            if (isset($source[0])) {
+                $compiled .= '// '.str_replace(array("\r\n", "\r", "\n"), "\n// ", $source)."\n";
```

Comment:
yes we can :) but why? getDebugInfo does not dependent on debug, and currently, the debug flag has no impact on the compile stage